products/cloudbuild/api.yaml

```diff
@@ -228,6 +233,93 @@ objects:
         description: |
           Contents of the build template. Either a filename or build template must be provided.
         properties:
+          - !ruby/object:Api::Type::NestedObject
+            name: 'source'
+            description: |
+              The location of the source files to build.
+            properties:
+              - !ruby/object:Api::Type::NestedObject
+                name: 'storageSource'
+                exactly_one_of:
+                  - storageSource
```

Comment:
Unfortunately we can't do `exactly_one_of` on elements of lists: https://github.com/hashicorp/terraform-plugin-sdk/issues/470. I'd recommend just documenting the behavior (we actually already do that elsewhere in this resource: https://github.com/GoogleCloudPlatform/magic-modules/blob/master/products/cloudbuild/terraform.yaml#L38-L40)

Reply:
Great catch - [addressed here](https://github.com/GoogleCloudPlatform/magic-modules/pull/3627/commits/67d19d9454c48c262e8ea7a53eaa0da999a75760)

I copied the approach for the `branchName`, `tagName`, and `commitSHA` from the other ones in the file (not overriding, just putting into the description).

Reply:
Looks like it's still here. Check out the generated code as well to see what it ends up doing: https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-3627-old..auto-pr-3627#diff-f0c51a3d504e195b74aeb122b2747c9eR318.

Reply:
Okay explicitly [removed here](https://github.com/GoogleCloudPlatform/magic-modules/pull/3627/commits/75a82179e08ddd88c7b93eed58997aed1fbefb54)